features/polyExplorer/src/context/explorer-context.jsx

```diff
@@ -195,6 +200,8 @@ export const ExplorerProvider = ({ children }) => {
     }
 
     function updatePodNavigation() {
+        // This is a temporary fix - when the HTRT is not full size anymore it should not change the title any longer
+        if (popUp) return;
```

Comment:
This is the only part I can't really explain. Also, line 236 does `!popup` which in that case might be always false. At any case, maybe verify if this is really what you want, and there's not a bit of logic below (beyond the `if`) that might actually be needed.

Reply:
If a screen title is set by a popup it would be overwritten immediately after, therefore the escaping. With the always false !popUp you are absolutely right ofc

Reply:
I would prefer putting the logic for the screen title here with the other, but we only have a component here and not type which makes it much easier than calling constructor. Would you prefer checking constructor and having the logic centralized?

Reply:
But createPopUp is called only if `navigation.firstRun` is false. In general, the problem is that this function has 3 different responsibilities: setting title, setting actions and setting active actions.

Reply:
Also, serving three different routes, using three global state variables... It's a complicated function.

Reply:
I mean I can refactor it to be more clear. Yet that doesn't answer my question ;)

Reply:
> I would prefer putting the logic for the screen title here with the other, but we only have a component here and not type which makes it much easier than calling constructor. Would you prefer checking constructor and having the logic centralized?

Centralized logic is always easier to understand and refactor. But all this is out of scope, I'd say.

Reply:
Although that check could maybe be done at invocation instead of inside the function itself?

Reply:
> I mean I can refactor it to be more clear. Yet that doesn't answer my question ;)

Well, refactoring would possibly be outside the scope; but what could be in scope is getting that check out of the function and into wherever it's invoked. That would take one responsibility out of the function (and leave the same number they had before, which were already maybe too many)

Reply:
Yeah it's what happens when theres just off-rule cases being added